src/main/java/com/aws/greengrass/builtin/services/lifecycle/LifecycleIPCEventStreamAgent.java

```diff
@@ -249,6 +265,32 @@ public void handleStreamEvent(EventStreamJsonMessage streamRequestEvent) {
         }
     }
 
+    /**
+     * Defer a component update.
+     *
+     * @param request     DeferComponentUpdateRequest object
+     * @param serviceName nam of the service deferring the update
+     * @throws InvalidArgumentsError if service name or deployment id inputs are invalid
+     */
+    public void deferComponentUpdate(DeferComponentUpdateRequest request, String serviceName) {
+        if (!componentUpdateListeners.containsKey(serviceName)) {
```

Comment:
This should be `componentUpdateListenersInternal`, right?

Reply:
I need to check both maps, will update